### ISSUE TITLE
Wrong timezone in calendar after switch from CET to CEST

### ISSUE THREAD

**Issue by hrk** [IsmaelMartinez/teams-for-linux#800]
**Describe the bug**
After the (local and automated) switch from CET to CEST (Central Europe, Timezone is Europe/Rome) the calendar still shows all meetings with CET timezone, i.e. with the timings off by one hour. I don't know if this is similar to #335 
I don't see any way to set the timezone in the client, so I guess it should be reading the one from my system.

**Comment by jijojosephk** [comment]
Would you mind adding some details like system info, package etc.

**Comment by jijojosephk** [comment]
Is time zone updated when you restart the app?

**Comment by hrk** [comment]
You are right, sorry. The timezone doesn't get updated when I restart the app - indeed it's been one month of restarting the app every day after the last sunday of March. :)
I'm currently using 1.0.51, built from source (yarn start from cloned repo), on Gentoo Linux. Last time I updated the sources, the build broke due to the new dependencies. I will try it again after the workday is over.

**Comment by jijojosephk** [comment]
Thanks. Sometimes we're helpless when the issue might be platform specific? Do you think you can have the flatpak in Gentoo?

**Comment by hrk** [comment]
I see that the latest is 1.0.83, released a few hours ago. I will try both that one from sources and the AppImage and report back.

**Comment by jijojosephk** [comment]
What do you see when you execute this? Do that in 2 places. 

1. click debug from menu then choose console.
2. Open any other browser open developer tool console.

![image](https://user-images.githubusercontent.com/11662148/236199848-37e2c972-d2f2-44cd-89d3-5a070daf35d8.png)

**Comment by hrk** [comment]
teams-for-linux (latest develop): Thu May 04 2023 13:20:55 GMT+0100 (GMT+01:00)
firefox: Date Thu May 04 2023 14:21:14 GMT+0200 (Ora legale dell’Europa centrale)

FWIW:
luca@E0557546 ~ $ cat /etc/timezone 
Europe/Rome
luca@E0557546 ~ $ ls -lha /etc/localtime 
lrwxrwxrwx 1 root root 33 21 Jul  2021 /etc/localtime -> ../usr/share/zoneinfo/Europe/Rome

**Comment by jijojosephk** [comment]
Like firefox can you try google chrome?

**Comment by jijojosephk** [comment]
if google chrome and app has the same value, probably the issue with chromium. Not with this app.

**Comment by hrk** [comment]
I tried running the simplest electronjs demo I could find (the simple-samples/activity-monitor), and the code sample (new Date() in console) displays the correct time.

**Comment by hrk** [comment]
TBF: opening https://teams.microsoft.com/ in a new Google Chrome window shows calendar entries in yet another timezone (GMT), but this could really be like #335 😆

**Comment by jijojosephk** [comment]
What
```js
new Date()
```
Gives you in google chrome?

**Comment by hrk** [comment]
in Google Chrome I get the date in GMT (so right now, 15:28 local time, it shows 13:28 GMT)

**Comment by hrk** [comment]
Some further experiments and results.
Just for clarity and to simplify the descriptions below: I'm in CEST (Central Europe Summer Time, which is GMT+2), I was in CET (Central Europe Time, GMT+1) until the last sunday of March.

No customization or other efforts or non standard configurations: new Date() in console:
- Firefox: localtime (same as clock in taskbar) = GMT+2
- Teams-for-Linux (AppImage or git clone): GMT+1
- Google Chrome: GMT

Debug/Tools -> Sensors -> Override location (setting Berlin, which is in the same timezone) -> new Date() in console:
- all three give the correct time (GMT+2) however the calendar keeps the wrong visualization (both in teams-for-linux and Google Chrome).

I don't know if this can be helpful.

**Comment by jijojosephk** [comment]
I expect `google-chrome` and `teams-for-linux` to behave the same. But this is weird!

**Comment by hrk** [comment]
TBH electronjs uses a bundled chromium (depending on the specific version of electronjs) which is different from my system's Google Chrome.

**Comment by hrk** [comment]
Unfortunately (for me) it's an upstream bug in Node (https://github.com/nodejs/node/issues/37271) which depends on upstream's ICU (https://unicode-org.atlassian.net/browse/ICU-13694). There is a patch (https://github.com/unicode-org/icu/pull/2213) which hasn't been merged yet. I've currently solved by prepending yarn start with a forced TZ variable.
